src/Native/System.Security.Cryptography.Native/pal_ssl.cpp

```diff
@@ -206,27 +216,27 @@ static CipherAlgorithmType MapCipherAlgorithmType(const char* encryption, size_t
 
 static ExchangeAlgorithmType MapExchangeAlgorithmType(const char* keyExchange, size_t keyExchangeLength)
 {
-    if (strncmp(keyExchange, "RSA", keyExchangeLength) == 0)
+    if (StringSpanEquals(keyExchange, "RSA", keyExchangeLength))
         return ExchangeAlgorithmType::RsaKeyX;
-    if (strncmp(keyExchange, "DH/RSA", keyExchangeLength) == 0)
+    if (StringSpanEquals(keyExchange, "DH/RSA", keyExchangeLength))
         return ExchangeAlgorithmType::DiffieHellman;
-    if (strncmp(keyExchange, "DH/DSS", keyExchangeLength) == 0)
+    if (StringSpanEquals(keyExchange, "DH/DSS", keyExchangeLength))
         return ExchangeAlgorithmType::DiffieHellman;
-    if (strncmp(keyExchange, "DH", keyExchangeLength) == 0)
+    if (StringSpanEquals(keyExchange, "DH", keyExchangeLength))
         return ExchangeAlgorithmType::DiffieHellman;
-    if (strncmp(keyExchange, "KRB5", keyExchangeLength) == 0)
+    if (StringSpanEquals(keyExchange, "KRB5", keyExchangeLength))
         return ExchangeAlgorithmType::SSL_kKRB5;
-    if (strncmp(keyExchange, "ECDH/RSA", keyExchangeLength) == 0)
+    if (StringSpanEquals(keyExchange, "ECDH", keyExchangeLength))
+        return ExchangeAlgorithmType::SSL_ECDHE;
```

Comment:
O...kay.  Well, if we match Windows now that's what's really important.

It'd be nice if we had some tests for this; but I understand the complexity involved in doing so.